Web_Music/Controllers/UserController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using AutoMapper;
+using BusinessLayer.Models;
+using BusinessLayer.Services.Interface;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using Web_Music.Models;
+
+namespace Web_Music.Controls
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class UserController: Controller
+    {
+        private readonly IUserService _userService;
+        private IMapper _mapper;
+        public UserController(IUserService userService, IMapper mapper)
+        {
+            _userService = userService;
+            _mapper = mapper;
+        }
+        [Route("Get/{id:int}")]
+        [HttpGet]
+        [ProducesResponseType(typeof(UserResponseModel), StatusCodes.Status200OK)]
+        public IActionResult GetById([FromRoute] int id)
+        {
+            try
+            {
+                var user = _userService.GetUser(id);
+                var getUser = _mapper.Map<UserResponseModel>(user);
+                return Ok(getUser);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex);
+            }
+        }
+        [Route("GetAll")]
+        [HttpGet]
+        [ProducesResponseType(typeof(UserResponseModel), StatusCodes.Status200OK)]
+        public IActionResult GetAllUsers()
+        {
+            try
+            {
+                var users = _userService.GetAllUser();
```

Comment:
_userService.GetAllUsers();

Reply:
before mapping users, check if the collection is not empty. If it is empty, return NotFound()